staging/src/k8s.io/api/events/v1/types.go

```diff
@@ -76,8 +72,8 @@ type Event struct {
 
 	// type is the type of this event (Normal, Warning), new types could be added in the future.
 	// It is machine-readable.
-	// +optional
-	Type string `json:"type,omitempty" protobuf:"bytes,11,opt,name=type"`
+	// This field cannot be empty for new Events.
+	Type string `json:"type" protobuf:"bytes,11,opt,name=type"`
```

Comment:
restore the omitempty, core/v1 events can omit this field and so a reader of events.k8s.io/v1 events could still encounter an object that omitted this field

Reply:
got it. (damn, I m making a lot of mistakes here 🙈  )